### DISCUSSION TITLE
File downloaded instead of redirecting App proxy

### DISCUSSION THREAD

**Discussion by sandeeppangeni** [osiset/laravel-shopify#674] [Q&A]
Need help

*For bug reporting only! If you're posting a feature request or discussion, please ignore.*

## Expected Behavior
 I have setup the app proxy as 
 
![image](https://user-images.githubusercontent.com/37801127/105982471-6a196000-60bf-11eb-94cf-434f6ca9aa5e.png)

And given the routes as

Route::get('/cartmade_payment', function () {
   return response('Hello, world!')->withHeaders(['Content-Type' => 'application/liquid']);
})->middleware('auth.proxy');

But it is downloading the file instead of redirecting

.htaccess file
`<IfModule mod_rewrite.c>
    <IfModule mod_negotiation.c>
        Options -MultiViews -Indexes
    </IfModule>

    RewriteEngine On

    # Handle Authorization Header
    RewriteCond %{HTTP:Authorization} .
    RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]

    # Redirect Trailing Slashes If Not A Folder...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_URI} (.+)/$
    RewriteRule ^ %1 [L,R=301]

    # Send Requests To Front Controller...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_FILENAME} !-f
    RewriteRule ^ index.php [L]
</IfModule>

# Set the “ea-php74” package as the default “PHP” programming language.
<IfModule mime_module>
  AddHandler application/x-httpd-ea-php74 .php .php7 .phtml
</IfModule>
`

### Steps to Reproduce
when I change the .htaccess file as below, I got 500 internal server.
# Redirect Trailing Slashes If Not A Folder...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_URI} (.+)/$
    RewriteRule ^ %1 [L,R=301]
Changes to

# Redirect Trailing Slashes If Not A Folder...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_URI} (.+)/$
    RewriteRule !^/?cartmade_payment*$ ^ %1 [L,R=301]

## Failure Information

![image](https://user-images.githubusercontent.com/37801127/105983348-b1ecb700-60c0-11eb-954b-50460f902601.png)

### Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

"php": "^7.3|^7.4|^8.0",
        "fideloper/proxy": "^4.4",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.0.1",
        "laravel/framework": "^8.12",
        "laravel/tinker": "^2.5",
        "osiset/basic-shopify-api": "^10.0",
        "osiset/laravel-shopify": "^15.1"

* Laravel Version:  8.8
* PHP Version: 8.0

**Comment by moonpetaldev** [comment]
Comment out these lines in your .htaccess, by adding a # to the start. It stops the 301 redirect which causes Shopify to download the file.

> `# Redirect Trailing Slashes If Not A Folder...`
> `RewriteCond %{REQUEST_FILENAME} !-d`
> `RewriteCond %{REQUEST_URI} (.+)/$`
> `RewriteRule ^ %1 [L,R=301]`
